tests/unittests/CoreFoundation/CFCppBaseTests.cpp

```diff
@@ -0,0 +1,101 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <TestFramework.h>
+#include <CFCppBase.h>
+
+struct __CFBoringClass : CoreFoundation::CppBase<__CFBoringClass> {};
+
+struct __CFLifetimeCounter : CoreFoundation::CppBase<__CFLifetimeCounter> {
+    int& ctorCountReference;
+    int& dtorCountReference;
+
+    __CFLifetimeCounter(int& ctorCounter, int& dtorCounter) : ctorCountReference(ctorCounter), dtorCountReference(dtorCounter) {
+        ++ctorCountReference;
+    }
+
+    ~__CFLifetimeCounter() {
+        ++dtorCountReference;
+    }
+};
+
+struct __CFSubLifetimeCounter : CoreFoundation::CppBase<__CFSubLifetimeCounter, __CFLifetimeCounter> {
+    __CFSubLifetimeCounter(int& ctorCounter, int& dtorCounter) : Parent(ctorCounter, dtorCounter) {
```

Comment:
This tests a few things: Members expressible only through C++ semantics work properly, constructors and destructors get called properly, and inherited members do not lose their magic.